src/types.ts

```diff
@@ -0,0 +1,70 @@
+export interface AddressInterface {
+  confidentialAddress: string;
+  blindingPrivateKey: string;
+  derivationPath?: string;
+}
+
+export interface SignedMessage {
+  signature: SignatureBase64;
+  address: NativeSegwitAddress;
+  publicKey: ECPublicKey;
+}
+
+export enum TxStatusEnum {
+  Confirmed = 1,
+  Pending = 0,
+}
+
+export interface Transaction {
+  txId: string;
+  status: TxStatusEnum;
+  fee: number;
+  transfers: Array<{ asset: string; amount: number }>;
+  explorerURL: string;
+  blocktimeMs: number;
+}
+
+export interface Utxo {
+  txid: string;
+  vout: number;
+  asset?: string;
+  value?: number;
+}
+
+export interface Balance {
+  asset: {
+    assetHash: string;
+    ticker?: string;
+    name?: string;
+    precision: number;
+  };
+  amount: number;
+}
+
+// add an OP_RETURN output
+export interface DataRecipient {
```

Comment:
even OP_RETURN should have asset and value (is actually the way to burn assets on Liquid)